truss/remote/truss_remote.py

```diff
@@ -142,6 +142,13 @@ def authenticate(self) -> dict:
         """
         return {}
 
```

Comment:
All the new types in this file are defined to satisfy the type checks. Can evolve the next time we have a truss remote

Reply:
Can you satisfy type checks by adding these methods to the Baseten remote subclass? The `TrussRemote` class is ideally a very minimal abstract class that provides an interface for all remotes that accept trusses. Adding log / development version functionality to the TrussRemote class breaks this as other remotes, such as SageMaker, may not have such abstractions.

Reply:
Short answer is -- no

The truss "cli.py" push / watch functions expect a generic remote object, not a Baseten remote object. So we can only call methods in those functions that exist on the `TrussRemote` class. Our options are either:
* Change `truss push` and `truss watch` to only work with Baseten remotes (which means these commands won't work with sagemaker & other remotes)
* Remove the viewing logs and truss watch functionality from the CLI

I think neither of these options are great. Can we go w/ this for now, and for other remotes simply raise `NotImplemented`? I think it's hard to have these conversations without knowing what the other remotes would be like / what `truss push` looks like in that world.

Reply:
Approved, we can revisit when a new implementation is added.